nxdrive/objects.py

```diff
@@ -141,86 +146,50 @@ class NuxeoDocumentInfo:
         """Convert Automation document description to NuxeoDocumentInfo"""
-        props = doc["properties"]
-        name = props["dc:title"]
-        folderish = "Folderish" in doc["facets"]
         try:
-            last_update = datetime.strptime(
-                doc["lastModified"], "%Y-%m-%dT%H:%M:%S.%fZ"
-            )
-        except ValueError:
-            # no millisecond?
-            last_update = datetime.strptime(doc["lastModified"], "%Y-%m-%dT%H:%M:%SZ")
+            root = doc["root"]
+            uid = doc["uid"]
+            path = doc["path"]
+            props = doc["properties"]
+            name = unicodedata.normalize("NFC", props["dc:title"])
+            folderish = "Folderish" in doc["facets"]
+            modified = doc["lastModified"]
+        except KeyError:
+            raise DriveError(f"This document is missing mandatory information: {doc}")
 
-        blobs: Dict[str, Blob] = {}
+        last_update = parser.parse(modified)
 
-        if not folderish:
-            main_blob = props.get("file:content")
-            attachments = props.get("files:files", [])
-            note = props.get("note:note")
-
-            if main_blob:
-                blobs["file:content"] = Blob.from_dict(main_blob)
-
-            for idx, attachment in enumerate(attachments):
-                blobs[f"files:files/{idx}/file"] = Blob.from_dict(attachment["file"])
-
-            if note:
-                m = hashlib.md5()
-                m.update(note.encode("utf-8"))
-                digest = m.hexdigest()
-                digest_algorithm = "md5"
-                ext = ".txt"
-                mime_type = props.get("note:mime_type")
-                if mime_type == "text/html":
-                    ext = ".html"
-                elif mime_type == "text/xml":
-                    ext = ".xml"
-                elif mime_type == "text/x-web-markdown":
-                    ext = ".md"
-                if not name.endswith(ext):
-                    name += ext
-
-                blobs["note:note"] = Blob.from_dict(
-                    {
-                        "name": name,
-                        "digest": digest,
-                        "digestAlgorithm": digest_algorithm,
-                        "length": len(note),
-                        "mime-type": mime_type,
-                        "data": note,
-                    }
-                )
+        blobs = {} if folderish else NuxeoDocumentInfo._parse_blobs(props)
 
         # Lock info
         lock_owner = doc.get("lockOwner")
         lock_created = doc.get("lockCreated")
-        if lock_created is not None:
+        if lock_created:
             lock_created = parser.parse(lock_created)
 
         # Permissions
         permissions = doc.get("contextParameters", {}).get("permissions", None)
 
         # Trashed
-        is_trashed = doc.get("isTrashed", doc["state"] == "deleted")
+        is_trashed = doc.get("isTrashed", doc.get("state") == "deleted")
```

Comment:
Perhaps using `doc.get("isTrashed", doc.get("state", "") == "deleted")` to prevent compring `None == "deleted"`?